components/brave_vpn/browser/connection/win/brave_vpn_helper/brave_vpn_helper_state.cc

```diff
@@ -18,46 +18,51 @@
 namespace brave_vpn {
 
 namespace {
-// Helper service has 3 fail actions configured to autorestart the service if
-// crashed. The check happens before the service started and counter set to 1,
-// thus we calculate attempts from 0 -> 2.
-const int kHelperServiceFailActionsNumber = 2;
 
 HRESULT HRESULTFromLastError() {
   const auto error_code = ::GetLastError();
   return (error_code != NO_ERROR) ? HRESULT_FROM_WIN32(error_code) : E_FAIL;
 }
 
-int GetServiceLaunchCounterValue() {
-  base::win::RegKey key(HKEY_LOCAL_MACHINE,
-                        brave_vpn::kBraveVpnHelperRegistryStoragePath,
-                        KEY_READ);
-  if (!key.Valid()) {
-    LOG(ERROR) << "Failed to read the successful launch counter";
-    return 0;
+}  // namespace
+
+bool IsBraveVPNHelperServiceInstalled() {
+  ScopedScHandle scm(::OpenSCManager(nullptr, nullptr, SC_MANAGER_CONNECT));
+  if (!scm.IsValid()) {
+    VLOG(1) << "::OpenSCManager failed. service_name: "
+            << brave_vpn::GetVpnServiceName() << ", error: " << std::hex
+            << HRESULTFromLastError();
+    return false;
   }
-  DWORD launch = 0;
-  key.ReadValueDW(kBraveVpnHelperLaunchCounterValue, &launch);
-  return launch;
-}
+  ScopedScHandle service(::OpenService(
+      scm.Get(), brave_vpn::GetVpnServiceName().c_str(), SERVICE_QUERY_STATUS));
 
-}  // namespace
+  // Service registered and has not exceeded the number of auto-configured
+  // restarts.
+  return service.IsValid();
+}
 
-bool IsBraveVPNHelperServiceLive() {
+bool IsBraveVPNHelperServiceRunning() {
   ScopedScHandle scm(::OpenSCManager(nullptr, nullptr, SC_MANAGER_CONNECT));
   if (!scm.IsValid()) {
-    LOG(ERROR) << "::OpenSCManager failed. service_name: "
-               << brave_vpn::GetVpnServiceName() << ", error: " << std::hex
-               << HRESULTFromLastError();
+    VLOG(1) << "::OpenSCManager failed. service_name: "
```

Comment:
Something to consider - for the service there won't really be a proper stdout/stderr, right?

We may consider writing to eventvwr. Not sure if Chromium has an example where they do that for crash reporting or updating (something in a service)

Reply:
Added support for `--log-file` command line to redirect logging to a file.